### PR TITLE
8267543: Post JEP 411 refactoring: security

### PR DESCRIPTION
**Pull request by wangweij**
For all modified files in #4073 having "security", "crypto", or "ssl" in their names. Codes are refactored to bring a `@SuppressWarning` annotation nearer to the deprecated API usage and also shrink the size of its target.

I'll add a copyright year update commit before integration.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8267543](https://bugs.openjdk.java.net/browse/JDK-8267543): Post JEP 411 refactoring: security

### Reviewers
 * [Sean Mullan](https://openjdk.java.net/census#mullan) (@seanjmullan - **Reviewer**) ⚠️ Review applies to 176f44e841600e029764b81e5148e8df41f85205

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4172/head:pull/4172` \
`$ git checkout pull/4172`

Update a local copy of the PR: \
`$ git checkout pull/4172` \
`$ git pull https://git.openjdk.java.net/jdk pull/4172/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4172`

View PR using the GUI difftool: \
`$ git pr show -t 4172`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4172.diff">https://git.openjdk.java.net/jdk/pull/4172.diff</a>

</details>

### CODE REVIEW

File: src/java.security.jgss/share/classes/sun/security/jgss/krb5/Krb5Context.java [hunk 1]
```diff
@@ -706,6 +706,7 @@ public KerberosTicket run() throws Exception {
                                     tgt);
                         }
                         if (GSSUtil.useSubjectCredsOnly(caller)) {
+                            @SuppressWarnings("removal")
```
Comment (seanjmullan) [inline src/java.security.jgss/share/classes/sun/security/jgss/krb5/Krb5Context.java:709]: It looks like you missed removing the @SuppressWarnings("removal") from the initSecContext method above.

Comment (wangweij) [reply src/java.security.jgss/share/classes/sun/security/jgss/krb5/Krb5Context.java:711]: Oops. Fixed.

**Comment by seanjmullan** [comment]
Update copyright dates to 2021, if applicable.

**Comment by wangweij** [comment]
> Update copyright dates to 2021, if applicable.

Will do. Since this PR can only be integrated after its parent PR #4073 is integrated. I'll delay the copyright update by then.

**Review by seanjmullan — APPROVED** [review]

**Pull request closed without merging**